src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -34,6 +34,8 @@ public TypeFactory(OutputLibrary library)
 
         public CSharpType CreateType(InputType inputType) => inputType switch
         {
+            InputLiteralType literalType       => CreateType(literalType.LiteralValueType), //TODO -- need to support literal type with the value.
```

Comment:
This is not the correct type returned for literal or union. Check the generated `Thing.cs` which doesn't follow what we've discussed.

Reply:
Codegen need to handle how to generate the literal or Union, there I just add workaround to pass generation.